flow/include/flow/ThreadHelper.actor.h

```diff
@@ -785,6 +812,41 @@ typename std::enable_if<!allow_anonymous_future<T>::value, Future<T>>::type safe
 	return future;
 }
 
+// Specialization for Standalone<T> and Optional<Standalone<T>>
+template <class T>
+Future<Standalone<T>> safeThreadFutureToFuture(ThreadFuture<Standalone<T>>& threadFuture) {
+	Future<Standalone<T>> f = safeThreadFutureToFutureImpl(threadFuture);
+	if (BUGGIFY) {
+		return removeArenaFromStandalone(f);
+	}
+	return f;
+}
+
+template <class T>
+Future<Optional<Standalone<T>>> safeThreadFutureToFuture(ThreadFuture<Optional<Standalone<T>>>& threadFuture) {
+	Future<Optional<Standalone<T>>> f = safeThreadFutureToFutureImpl(threadFuture);
+	if (BUGGIFY) {
+		return removeArenaFromStandalone(f);
+	}
+	return f;
+}
+
+template <class T>
+Future<Standalone<T>> safeThreadFutureToFuture(Future<Standalone<T>>& future) {
+	if (BUGGIFY) {
+		return removeArenaFromStandalone(future);
+	}
+	return future;
+}
+
+template <class T>
+Future<Optional<Standalone<T>>> safeThreadFutureToFuture(Future<Optional<Standalone<T>>>& future) {
+	if (BUGGIFY) {
+		return removeArenaFromStandalone(future);
+	}
+	return future;
+}
```

Comment:
With these, the `allow_anonymous_future` bit may be redundant. Or perhaps better, you could put these alternate implementations into the two overloads above where `allow_anonymous_future` is false.

Reply:
I thought about this, and was unsure about it because I wonder whether `allow_anonymous_future == false` implies `Standalone` or `Optional<Standalone>`.

Reply:
`allow_anonymous_future` is currently declared false for those two types (and only those two types), and essentially what we'd be saying is that if we added another future type to this set then it should also implement the `removeArenaFromStandalone` function for that type.

Reply:
Make sense. Will update.